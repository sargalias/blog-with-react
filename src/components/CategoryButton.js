import React from 'react';

const CategoryButton = ({label, isActive=false, hoverColor}) => (
  <li
    className={`nav-item ${isActive && 'active'}`}
    style={{color: hoverColor}}
  >
    <a
      className="nav-link"
      href="/"
      onClick={(e) => e.preventDefault()}
    >
      {label} {isActive && <span className="sr-only">(current)</span>}
    </a>
  </li>
);

export default CategoryButton;
