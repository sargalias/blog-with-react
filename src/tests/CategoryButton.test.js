import React from 'react';
import { shallow, render } from 'enzyme';
import CategoryButton from "../components/CategoryButton";

test('CategoryButton renders correctly with no props', () => {
  const wrapper = shallow(<CategoryButton />);
  expect(wrapper).toMatchSnapshot();
});

test('CategoryButton renders correctly with props', () => {
  const wrapper = shallow(<CategoryButton label='test' isActive={true} hoverColor='orange' />);
  expect(wrapper).toMatchSnapshot();
});

test('CategoryButton correctly adds hoverColor prop to inline styles', () => {
  const hoverColor = 'orange';
  const wrapper = render(<CategoryButton hoverColor={hoverColor} />);
  expect(wrapper.css('color')).toBe(hoverColor);
});

test('CategoryButton has correct label', () => {
  const label = 'test text';
  const re = new RegExp(label);
  const wrapper = render(<CategoryButton label={label} />);
  expect(wrapper.text()).toMatch(re);
});

test('CategoryButton is set correctly when isActive=true', () => {
  const isActive = true;
  const expected = ' (current)';
  const wrapper = render(<CategoryButton isActive={isActive} />);
  expect(wrapper.text()).toBe(expected);
});

test('CategoryButton is set correctly when isActive=false', () => {
  const isActive = false;
  const expected = ' ';
  const wrapper = render(<CategoryButton isActive={isActive} />);
  expect(wrapper.text()).toBe(expected);
});
