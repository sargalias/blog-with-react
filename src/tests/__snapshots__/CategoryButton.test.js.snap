// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryButton renders correctly with no props 1`] = `
<li
  className="nav-item false"
  style={
    Object {
      "color": undefined,
    }
  }
>
  <a
    className="nav-link"
    href="/"
    onClick={[Function]}
  >
     
  </a>
</li>
`;

exports[`CategoryButton renders correctly with props 1`] = `
<li
  className="nav-item active"
  style={
    Object {
      "color": "orange",
    }
  }
>
  <a
    className="nav-link"
    href="/"
    onClick={[Function]}
  >
    test
     
    <span
      className="sr-only"
    >
      (current)
    </span>
  </a>
</li>
`;
